### DISCUSSION TITLE
Centering a container with uncertain size

### DISCUSSION THREAD

**Discussion by kuku1234us** [tailwindlabs/tailwindcss#12302] [Help] [answered]
Dear all, I'm trying to center  a list of cards in a flex box while having the items in the last row align with the rest of the rows. Meaning, center the red cards in the middle of the screen.

![Cards](https://github.com/tailwindlabs/tailwindcss/assets/120145807/d7caf100-e3d6-4916-9331-71e4c9316079)

I tried wrapping the red cards with a container with w-fit as show below:
```
<div className="flex justify-center items-center w-full p-5">
      <div
        id="innerBox"
        className="border border-blue-500 m-auto flex flex-wrap w-fit gap-5 items-center justify-start p-5"
      >
        <div className="w-[350px] h-[100px] redBorder">Card 1</div>
        <div className="w-[350px] h-[100px] redBorder">Card 2</div>
        <div className="w-[350px] h-[100px] redBorder">Card 3</div>
        <div className="w-[350px] h-[100px] redBorder">Card 4</div>
        <div className="w-[350px] h-[100px] redBorder">Card 5</div>
      </div>
</div>
```

As shown in the screen shot above, the innerBox with the blue border is taking up more width as it should, resulting in the cards justifying to the left of the screen instead of the center.

How can I center the cards on the screen (while keeping the odd items in the last row aligned with the rest)? Thanks in advance.

**Comment by wongjn** [comment]
You could consider applying `justify-content: center` instead of `justify-content: flex-start` to `#innerBox: https://play.tailwindcss.com/oVOUUJjQtm

**Reply by wongjn** [reply]
> @wongjn Applying justify-center was my first instinct too. However, the last row will be out of alignment with the rest of the rows as shown below:[^1]

I believe that now satisfies your requirements, as per your original post:

> Meaning, center the red cards in the middle of the screen.

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/12302#discussioncomment-7404264

**Reply by kuku1234us** [reply]
I am sorry if my original description of the question is not 100%. Using justify-center straight up would result in the cards in the last row out of alignment with the rest of the rows. I am looking to achieve the effect where the block of cards as a whole is centered in the parent container, while the cards in the last row remains aligned with the rest of the cards, as shown in the following image:

![Cards03](https://github.com/tailwindlabs/tailwindcss/assets/120145807/3f7f4c8b-12bd-4da5-9d97-61e0506ff7b0)

I am trying to avoid using a grid since I'd have to define the number of columns for every screen size. Thanks for helping in advance!

**Reply by wongjn** [reply, marked as answer]
Consider using a grid layout **without** defining the number of columns for every screen size: https://play.tailwindcss.com/lSQSAuXKUQ

**Reply by kuku1234us** [reply]
Ahh!  I think that does the trick!  Thanks so much Justin!

**Comment by kuku1234us** [comment]
@wongjn Applying justify-center was my first instinct too. However, the last row will be out of alignment with the rest of the rows as shown below:

![Cards02](https://github.com/tailwindlabs/tailwindcss/assets/120145807/ed540534-8c3c-4e6f-a5ba-577401e32676)

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/12302#discussioncomment-7405500 to keep the reply chain together.
